src/Persistence/MassTransit.EntityFrameworkCoreIntegration/EntityFrameworkCoreIntegration/Saga/DbContextSagaRepositoryContext.cs

```diff
@@ -159,18 +159,18 @@ public class DbContextSagaRepositoryContext<TSaga> :
         where TSaga : class, ISaga
     {
         readonly DbContext _dbContext;
+        readonly ILoadQueryExecutor<TSaga> _loadExecutor;
 
-        public DbContextSagaRepositoryContext(DbContext dbContext, CancellationToken cancellationToken)
+        public DbContextSagaRepositoryContext(ILoadQueryExecutor<TSaga> loadExecutor, DbContext dbContext, CancellationToken cancellationToken)
             : base(cancellationToken)
         {
+            _loadExecutor = loadExecutor;
             _dbContext = dbContext;
         }
 
         public Task<TSaga> Load(Guid correlationId)
         {
-            return _dbContext.Set<TSaga>()
-                .AsNoTracking()
-                .SingleOrDefaultAsync(x => x.CorrelationId == correlationId);
+            return _loadExecutor.Load(_dbContext, correlationId, CancellationToken);
```

Comment:
This needs to be AsNoTracking() though.

Reply:
It is, look what is passed in this class